src/components/link/link.component.tsx

```diff
@@ -162,6 +162,12 @@ export const Link = React.forwardRef<
       );
     };
 
+    useEffect(() => {
+      if (disabled || !(href && onClick)) {
```

Comment:
**issue**: we have a case with `Pager` where `Link` rendered without `onClick` or `href`, so the state will be set to false on every render:

https://github.com/user-attachments/assets/d845e634-6f4b-4be1-be13-8ab21a641cd3

```suggestion
      if (disabled || !(href || onClick)) {
```

Reply:
Yep you're right I rushed it and forgot to change the operator to OR as originally had it as `!href && !onClick`. The fact there's an `onClick` and no `href `is being negated from false to true, I don't think there's a point where Pager links don't have an onClick, the types say they're optional but they're always passed as we wrap the prop callbacks in useCallback then pass it etc but it's irrelevant here